cpp/bench/ann/src/hnswlib/hnswlib_wrapper.h

```diff
@@ -192,7 +175,10 @@ void hnsw_lib<T>::search(
     get_search_knn_results(query + i * dim_, k, indices + i * k, distances + i * k);
   };
   if (bench_mode_ == Mode::kLatency && num_threads_ > 1) {
-    thread_pool_->submit(f, batch_size);
+#pragma omp parallel for num_threads(num_threads_)
```

Comment:
Can we introduce a conditional here so that the thread creation is done only when batch_size is greater than 1?
`#pragma omp parallel if(batch_size > 1) for num_threads(num_threads_)`